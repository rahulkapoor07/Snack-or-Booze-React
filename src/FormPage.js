import React, {useState} from "react";
import {useHistory} from "react-router-dom";
import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

const FormPage = ({collectFormData})=>{
    const INITIAL_STATE = {type: "",name : "", description : "",
                            recipe: "", serve : ""}
    const history = useHistory();
    
    const [formData, seFormData] = useState(INITIAL_STATE);

    const handleChange = (e) =>{
        const {name, value} = e.target;
        seFormData(data => ({...data, 
        [name] : value}))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        collectFormData({...formData});
        history.push("/");

    }

    return(
        <Form onSubmit={handleSubmit}>
            
        <FormGroup>
        <Label for="exampleSelect">Select</Label>
        <Input type="select" name="type" id="exampleSelect"
        value={formData.type} onChange={handleChange}>
          <option>None</option>
          <option>Snacks</option>
          <option>Drinks</option>
        </Input>
      </FormGroup>

      <FormGroup>
        <Label for="typeName">Name</Label>
        <Input type="text" name="name" id="typeName" 
        value = {formData.name} onChange={handleChange}
        placeholder="Name" />
      </FormGroup>

      <FormGroup>
        <Label for="typeDescription">Description</Label>
        <Input type="text" name="description" id="typeDescription" 
        value = {formData.description} onChange={handleChange}
        placeholder="Description" />
      </FormGroup>

      <FormGroup>
        <Label for="recipe">Recipe</Label>
        <Input type="text" name="recipe" 
        value={formData.recipe} onChange={handleChange}
        id="recipe" placeholder="Recipe" />
      </FormGroup>

      <FormGroup>
        <Label for="serve">Serve</Label>
        <Input type="textarea" name="serve" 
        value={formData.serve} onChange = {handleChange}
        id="serve" />
      </FormGroup>
      <Button>Submit</Button>
    </Form>
    )
}


export default FormPage;