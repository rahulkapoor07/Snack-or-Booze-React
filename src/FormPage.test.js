import React from "react";
import {MemoryRouter} from "react-router-dom";
import { render, fireEvent } from '@testing-library/react';
import FormPage from "./FormPage";

describe("App page testing", ()=>{
    test("rendering", ()=>{
        render(<MemoryRouter><FormPage /></MemoryRouter>);
    });

    test("snapshot", ()=>{
        const {asFragment} = render((<MemoryRouter><FormPage /></MemoryRouter>));
        expect(asFragment()).toMatchSnapshot();
    });

    test("Testing form", ()=>{
        const {getByText, getByLabelText} = render((<MemoryRouter initialEntries={["/add-new"]}><FormPage /></MemoryRouter>));
        expect(getByLabelText("Name")).toBeInTheDocument();
        const btn = getByText("Submit");
        expect(btn).toBeInTheDocument();
    });
});