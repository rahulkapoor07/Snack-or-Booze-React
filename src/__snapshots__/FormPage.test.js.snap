// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page testing snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="exampleSelect"
      >
        Select
      </label>
      <select
        class="form-control"
        id="exampleSelect"
        name="type"
      >
        <option>
          None
        </option>
        <option>
          Snacks
        </option>
        <option>
          Drinks
        </option>
      </select>
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="typeName"
      >
        Name
      </label>
      <input
        class="form-control"
        id="typeName"
        name="name"
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="typeDescription"
      >
        Description
      </label>
      <input
        class="form-control"
        id="typeDescription"
        name="description"
        placeholder="Description"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="recipe"
      >
        Recipe
      </label>
      <input
        class="form-control"
        id="recipe"
        name="recipe"
        placeholder="Recipe"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="serve"
      >
        Serve
      </label>
      <textarea
        class="form-control"
        id="serve"
        name="serve"
      />
    </div>
    <button
      class="btn btn-secondary"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
